apps/06.1Checkers.js

```diff
@@ -47,21 +53,91 @@ function Board() {
+    ];
+    var blackPositions = [
+      [5, 0], [5, 2], [5, 4], [5, 6], 
+      [6, 1], [6, 3], [6, 5], [6, 7], 
+      [7, 0], [7, 2], [7, 4], [7, 6],
+    ];
+    // a better way to loop through using for...of... loop
+    //for (var spot of whiteCheckers) {
+    //   var row = spot[0];
+    //   var column = spot[1];
+    //   var whiteChecker = new Checker('white');
+    //   this.grid[row][column] = whiteChecker;
+    // }
+    for (var i = 0; i <= 11; i++) {
+      //instantiate a white checker
+      var whiteChecker = new Checker('white');
+      //place the checker on the grid at the corresponding index 
+      this.grid[whitePositions[i][0]][whitePositions[i][1]] = whiteChecker;
+      //push the checkers into the checker array 
+      this.checkers.push(whiteChecker);
+      //do the same thing for the black checker 
+      var blackChecker = new Checker('black');
+      this.grid[blackPositions[i][0]][blackPositions[i][1]] = blackChecker;
+      this.checkers.push(blackChecker);
+    }
+  }
+  //help function for selecting checkers on board
+  this.selectChecker = function(checkerPosition) {
+    //get the row and column indexes of the position passed in 
+    var row = checkerPosition[0];
+    var column = checkerPosition[1];
+    //put the checker in the variable c 
+    var c = this.grid[row][column];
+    //nullify the checker on board
+    this.grid[row][column] = null;
+    return c;
+  }
+  //helper function for killing a checker 
+  this.killChecker = function(position) {
+    //get the checker using the helper function 
+    var checkerForKill = this.selectChecker(position);
+    //get the index of the checker to be killed 
+    var killPosition = this.checkers.indexOf(checkerForKill);
+    //remove the checker from the array 
+    this.checkers.splice(killPosition,1);
+    //assign the checker's position on the grid to null 
+    this.grid[position[0]][position[1]] = null;
+    }
+};
 
+function Game() {
   this.board = new Board();
-
   this.start = function() {
     this.board.createGrid();
-    // Your code here
+    // Call the createCheckers() function and map out all the checkers on the grid 
+    this.board.createCheckers();
   };
+  this.moveChecker = (start, end) => {
+    //select the checker and put it into the 'checker' variable 
+    var checker = this.board.selectChecker(start);
+    // this.board.grid[start[0]][start[1]] = null; 
+    // assign the checker to the end position 
+    this.board.grid[end[0]][end[1]] = checker;
+    // this.board.selectChecker(end) = checker;
+    // why didn't the above line work? said "invalid left-hand assignment"
+    // check if the vertical distance between the start and end positions is 2
+    if (Math.abs(end[0]-start[0]) === 2) {
+        //get the midpoint position by adding both coordinates up, and divide the result by 2 
+        var killPosition = [(parseInt(end[0]) + parseInt(start[0]))/2, (parseInt(end[1]) + parseInt(start[1]))/2];
```

Comment:
Nice killPosition logic. You could simplify this statement by simply parsing the start and end arguments directly. 

ex: `var killPosition = ((parseInt(end) + parseInt(start)) / 2).toString();